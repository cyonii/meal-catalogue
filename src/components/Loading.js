import PropTypes from 'prop-types';

const Loading = (props) => {
  const { variant } = props;
  let spinnerClass = 'spinner-border';

  if (variant.length > 0) {
    spinnerClass += ` spinner-border-${variant}`;
  }

  return (
    <div className="d-flex flex-column justify-content-center align-items-center mt-5" data-testid="loading">
      <div className={spinnerClass} role="status">
        <span className="visually-hidden sr-only">Loading...</span>
      </div>
    </div>
  );
};

Loading.propTypes = {
  variant: PropTypes.string,
};

Loading.defaultProps = {
  variant: '',
};

export default Loading;
