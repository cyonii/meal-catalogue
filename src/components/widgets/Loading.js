import PropTypes from 'prop-types';

const Loading = (props) => {
  const { size } = props;

  return (
    <div className="d-flex flex-column justify-content-center align-items-center mt-5" data-testid="loading">
      <i className="bi bi-lightning-charge text-secondary spin" style={{ fontSize: size }} />
    </div>
  );
};

Loading.propTypes = {
  size: PropTypes.string,
};

Loading.defaultProps = {
  size: '1.5rem',
};

export default Loading;
