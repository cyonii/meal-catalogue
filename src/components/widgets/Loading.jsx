import PropTypes from "prop-types";

const Loading = (props) => {
  const { size } = props;

  return (
    <div
      className="h-[calc(100vh-64px)] flex flex-col justify-center items-center mt-5"
      data-testid="loading"
    >
      <i
        className="bi bi-lightning-charge text-slate-500 animate-spin opacity-75"
        style={{ fontSize: size, animationDuration: "0.3s" }}
      />
      {/* <span class="animate-ping inline-flex h-[40px] w-[40px] rounded-full bg-sky-400 opacity-75"></span> */}
    </div>
  );
};

Loading.propTypes = {
  size: PropTypes.string,
};

Loading.defaultProps = {
  size: "1.5rem",
};

export default Loading;
