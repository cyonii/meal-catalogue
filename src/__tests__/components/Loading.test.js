import { render, screen, cleanup } from '@testing-library/react';
import Loading from '../../components/Loading';

describe('Loading', () => {
  afterEach(() => cleanup());

  it('should render correctly', async () => {
    render(<Loading />);

    const loadingEl = await screen.findByTestId('loading');
    const spinner = screen.getByRole('status');

    expect(loadingEl).toMatchSnapshot();
    expect(spinner).toBeInTheDocument();
    expect(screen.getByTestId('loading')).toBeInTheDocument();
  });

  it('should render with given variant', () => {
    render(<Loading variant="sm" />);

    expect(screen.getByRole('status')).toHaveClass('spinner-border-sm');
  });
});
