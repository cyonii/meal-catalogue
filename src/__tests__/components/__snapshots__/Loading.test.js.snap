// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should render correctly 1`] = `
<div
  class="d-flex flex-column justify-content-center align-items-center mt-5"
  data-testid="loading"
>
  <div
    class="spinner-border"
    role="status"
  >
    <span
      class="visually-hidden sr-only"
    >
      Loading...
    </span>
  </div>
</div>
`;
